testdata/schema_derivation_tests/order_by.yml

```diff
@@ -32,7 +41,15 @@ dataset:
       docs: []
     schema:
       bsonType: "object"
-      additionalProperties: true
+      required: [ "_id", "a" ]
+      additionalProperties: false
+      properties:
+        _id:
+          bsonType: "int"
+        a:
+          anyOf:
+            - bsonType: "int"
+            - bsonType: "null"
```

Comment:
This works with `!!str`?

Reply:
`  {
    _id: 'nullAndNonNull',
    type: 'collection',
    schema: {
      bsonType: 'object',
      required: [ '_id', 'a' ],
      additionalProperties: false,
      properties: {
        _id: { bsonType: 'int' },
        a: { anyOf: [ { bsonType: 'int' }, { bsonType: 'null' } ] }
      }
    },
    lastUpdated: ISODate('2025-05-29T02:55:09.119Z')
  },` 🤷‍♂️

Reply:
This is great news